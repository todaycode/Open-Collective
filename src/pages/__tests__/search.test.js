/* global it, expect, describe */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'

import Router from 'next/router';

import { MockSearchPage } from '../search';

describe('Search Page', () => {
  const props = {
    data: {},
    getLoggedInUser: () => Promise.resolve({}),
  };

  it('renders loading grid SVG', () => {
    const loadingProps = {
      ...props,
      data: {
        loading: true,
      },
    };

    const tree = renderer.create(<MockSearchPage {...loadingProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders empty results message', () => {
    const emptyResultsProps = {
      ...props,
      data: {
        search: {
          collectives: [],
        },
      },
      term: 'test',
    };

    const tree = renderer.create(<MockSearchPage {...emptyResultsProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders error message', () => {
    const errorProps = {
      ...props,
      data: {
        error: {
          message: 'Test error',
        },
      },
    };

    const tree = renderer.create(<MockSearchPage {...errorProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders search results with pagination', () => {
    const resultsProps = {
      ...props,
      data: {
        search: {
          collectives: [{
            slug: 'test-collective',
          }],
          limit: 20,
          offset: 0,
          total: 100,
        },
      },
      term: 'Test',
    };

    const tree = renderer.create(<MockSearchPage {...resultsProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls Router.push on form submission', () => {
    Router.push = jest.fn();

    const submitProps = {
      ...props,
      url: {
        pathname: '/search',
      },
    };
    const event = {
      preventDefault: jest.fn(),
      target: {
        q: {
          value: 'test',
        },
      },
    };

    const wrapper = shallow(<MockSearchPage {...submitProps} />);
    wrapper.find('form').simulate('submit', event);

    expect(event.preventDefault).toHaveBeenCalled();
    expect(Router.push).toHaveBeenCalledWith({
      pathname: '/search',
      query: {
        q: 'test',
      },
    });
  });
});
