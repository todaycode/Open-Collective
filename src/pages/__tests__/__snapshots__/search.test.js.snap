// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Page renders empty results message 1`] = `
.c1 {
  box-sizing: border-box;
  width: 100%;
}

.c2 {
  box-sizing: border-box;
  margin-top: 16px;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c5 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  box-sizing: border-box;
  width: 100%;
  padding-top: 16px;
  padding-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3.c3.c3 {
  border: none;
  border-bottom: 2px solid #46B0ED;
  border-radius: 0;
  box-shadow: none;
  display: block;
  font-family: lato;
  height: 3.4rem;
  padding: 0;
}

.c4.c4 {
  padding: 0 2rem;
}

.c0 {
  box-sizing: border-box;
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  max-width: 1200px;
}

@media screen and (min-width:40em) {
  .c5 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media screen and (min-width:52em) {
  .c5 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    width: 85%;
  }
}

<div>
  <header
    className="jsx-3598562734"
  >
    <div
      className="jsx-3598562734"
      id="top"
    />
    <div
      className="jsx-3593811691 loading TopBar"
    >
      <a
        className="jsx-3593811691"
        href="/"
        title="Go to Open Collective Homepage"
      >
        <img
          alt="Open Collective logo"
          className="jsx-3593811691 logo"
          height="40"
          src="/static/images/opencollective-icon.svg"
          width="40"
        />
      </a>
      <div
        className="jsx-3593811691 nav"
      >
        <ul
          className="jsx-3593811691 mediumScreenOnly"
        >
          <li
            className="jsx-3593811691"
          >
            <a
              className="jsx-3593811691 menuItem"
              href="/learn-more"
            >
              <span>
                How it works
              </span>
            </a>
          </li>
          <li
            className="jsx-3593811691"
          >
            <a
              className="jsx-3593811691 menuItem"
              href="/discover"
            >
              <span>
                discover
              </span>
            </a>
          </li>
          <li
            className="jsx-3593811691"
          >
            <a
              className="jsx-3593811691 menuItem"
              href="https://medium.com/open-collective"
            >
              <span>
                Blog
              </span>
            </a>
          </li>
        </ul>
        <div
          className="jsx-3593811691 separator"
        />
        <div
          className="jsx-2783800609 LoginTopBarProfileButton"
        >
          <div
            className="jsx-2904874222 TopBarProfileMenu "
            onClick={[Function]}
          >
            <div
              className="jsx-2904874222 LoginTopBarProfileButton-name desktopOnly"
            />
            <div
              className="jsx-2904874222 LoginTopBarProfileButton-caret"
            />
          </div>
        </div>
      </div>
    </div>
  </header>
  <main
    className="jsx-4138488456"
  >
    <div
      className="c0"
    >
      <div
        className="c1"
      >
        <form
          method="GET"
          onSubmit={[Function]}
        >
          <div
            className="form-group form-group-lg"
          >
            <label
              className="h1 control-label"
              htmlFor="search"
            >
              Search Open Collective
            </label>
            <div
              className="c2"
            >
              <input
                className="c3 form-control"
                defaultValue="test"
                id="search"
                name="q"
                placeholder="open source"
                type="search"
              />
              <button
                className="jsx-2820878800 jsx-219750138 Button blue c4"
                onClick={[Function]}
                type="submit"
              >
                <span
                  className="fa fa-search"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
      <div
        className="c5"
      >
        <div
          className="c6"
        >
          <p>
            <em>
              No collectives found matching your query: "
              test
              "
            </em>
          </p>
        </div>
      </div>
    </div>
  </main>
  <div
    className="jsx-568600484 Footer"
  >
    <div
      className="jsx-568600484 content"
    >
      <div
        className="jsx-568600484"
      >
        <object
          className="jsx-568600484 logo"
          data="/static/images/opencollectivelogo-footer.svg"
          height="20"
          type="image/svg+xml"
        />
      </div>
      <div
        className="jsx-568600484"
      >
        <nav
          className="jsx-568600484"
        >
          <ul
            className="jsx-568600484"
          >
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="/learn-more"
              >
                How It Works
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://opencollective.com/#opensource"
              >
                Use Cases
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://opencollective.com/faq"
              >
                FAQ
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://forum.opencollective.com"
              >
                Forum
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://medium.com/open-collective"
              >
                Blog
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                href="/tos"
                onClick={[Function]}
              >
                Terms Of Service
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://opencollective.com/about"
              >
                About
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="mailto:info@opencollective.com"
              >
                Contact
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</div>
`;

exports[`Search Page renders error message 1`] = `
<div
  className="jsx-3855972673 Error"
>
  <header
    className="jsx-3598562734"
  >
    <div
      className="jsx-3598562734"
      id="top"
    />
    <div
      className="jsx-3593811691 TopBar"
    >
      <a
        className="jsx-3593811691"
        href="/"
        title="Go to Open Collective Homepage"
      >
        <img
          alt="Open Collective logo"
          className="jsx-3593811691 logo"
          height="40"
          src="/static/images/opencollective-icon.svg"
          width="40"
        />
      </a>
      <form
        action="/search"
        className="jsx-3593811691 topbar-search-form"
        method="GET"
      >
        <div
          className="jsx-3593811691 topbar-search-container"
        >
          <input
            className="jsx-3593811691 topbar-search-input"
            name="q"
            placeholder="Search Open Collective"
            type="search"
          />
          <button
            className="jsx-3593811691 topbar-search-button"
          >
            <svg
              height={16}
              version="1.1"
              viewBox="10 10 30 30"
              width={16}
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                magnifying glass icon
              </title>
              <path
                d="M28.5404814,30.1903972 C26.7995948,31.5736628 24.5962981,32.4 22.2,32.4 C16.5666956,32.4 12,27.8333044 12,22.2 C12,16.5666956 16.5666956,12 22.2,12 C27.8333044,12 32.4,16.5666956 32.4,22.2 C32.4,24.5962981 31.5736628,26.7995948 30.1903972,28.5404814 L35.6582912,34.0083754 C36.1139029,34.4639871 36.1139029,35.2026796 35.6582912,35.6582912 C35.2026796,36.1139029 34.4639871,36.1139029 34.0083754,35.6582912 L28.5404814,30.1903972 Z M22.2,30 C26.507821,30 30,26.507821 30,22.2 C30,17.892179 26.507821,14.4 22.2,14.4 C17.892179,14.4 14.4,17.892179 14.4,22.2 C14.4,26.507821 17.892179,30 22.2,30 Z"
                fill="#FFFFFF"
                fillRule="nonzero"
                id="magnifying-glass"
              />
            </svg>
          </button>
        </div>
      </form>
      <div
        className="jsx-3593811691 nav"
      >
        <ul
          className="jsx-3593811691 mediumScreenOnly"
        >
          <li
            className="jsx-3593811691"
          >
            <a
              className="jsx-3593811691 menuItem"
              href="/learn-more"
            >
              <span>
                How it works
              </span>
            </a>
          </li>
          <li
            className="jsx-3593811691"
          >
            <a
              className="jsx-3593811691 menuItem"
              href="/discover"
            >
              <span>
                discover
              </span>
            </a>
          </li>
          <li
            className="jsx-3593811691"
          >
            <a
              className="jsx-3593811691 menuItem"
              href="https://medium.com/open-collective"
            >
              <span>
                Blog
              </span>
            </a>
          </li>
        </ul>
        <div
          className="jsx-3593811691 separator"
        />
        <div
          className="jsx-2783800609 LoginTopBarProfileButton"
        >
          <div
            className="jsx-2783800609 LoginTopBarProfileButton"
          >
            <a
              className="jsx-2783800609"
              href="/signin?next=about%3Ablank"
              onClick={[Function]}
            >
              <span>
                login
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </header>
  <main
    className="jsx-4138488456"
  >
    <h1
      className="jsx-3855972673"
    >
      Test error
    </h1>
  </main>
  <div
    className="jsx-568600484 Footer"
  >
    <div
      className="jsx-568600484 content"
    >
      <div
        className="jsx-568600484"
      >
        <object
          className="jsx-568600484 logo"
          data="/static/images/opencollectivelogo-footer.svg"
          height="20"
          type="image/svg+xml"
        />
      </div>
      <div
        className="jsx-568600484"
      >
        <nav
          className="jsx-568600484"
        >
          <ul
            className="jsx-568600484"
          >
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="/learn-more"
              >
                How It Works
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://opencollective.com/#opensource"
              >
                Use Cases
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://opencollective.com/faq"
              >
                FAQ
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://forum.opencollective.com"
              >
                Forum
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://medium.com/open-collective"
              >
                Blog
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                href="/tos"
                onClick={[Function]}
              >
                Terms Of Service
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://opencollective.com/about"
              >
                About
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="mailto:info@opencollective.com"
              >
                Contact
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</div>
`;

exports[`Search Page renders loading grid SVG 1`] = `
.c1 {
  box-sizing: border-box;
  width: 100%;
}

.c2 {
  box-sizing: border-box;
  margin-top: 16px;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c5 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  box-sizing: border-box;
  width: 100%;
  padding-top: 16px;
  padding-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3.c3.c3 {
  border: none;
  border-bottom: 2px solid #46B0ED;
  border-radius: 0;
  box-shadow: none;
  display: block;
  font-family: lato;
  height: 3.4rem;
  padding: 0;
}

.c4.c4 {
  padding: 0 2rem;
}

.c0 {
  box-sizing: border-box;
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  max-width: 1200px;
}

@media screen and (min-width:40em) {
  .c5 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media screen and (min-width:52em) {
  .c5 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    width: 85%;
  }
}

<div>
  <header
    className="jsx-3598562734"
  >
    <div
      className="jsx-3598562734"
      id="top"
    />
    <div
      className="jsx-3593811691 loading TopBar"
    >
      <a
        className="jsx-3593811691"
        href="/"
        title="Go to Open Collective Homepage"
      >
        <img
          alt="Open Collective logo"
          className="jsx-3593811691 logo"
          height="40"
          src="/static/images/opencollective-icon.svg"
          width="40"
        />
      </a>
      <div
        className="jsx-3593811691 nav"
      >
        <ul
          className="jsx-3593811691 mediumScreenOnly"
        >
          <li
            className="jsx-3593811691"
          >
            <a
              className="jsx-3593811691 menuItem"
              href="/learn-more"
            >
              <span>
                How it works
              </span>
            </a>
          </li>
          <li
            className="jsx-3593811691"
          >
            <a
              className="jsx-3593811691 menuItem"
              href="/discover"
            >
              <span>
                discover
              </span>
            </a>
          </li>
          <li
            className="jsx-3593811691"
          >
            <a
              className="jsx-3593811691 menuItem"
              href="https://medium.com/open-collective"
            >
              <span>
                Blog
              </span>
            </a>
          </li>
        </ul>
        <div
          className="jsx-3593811691 separator"
        />
        <div
          className="jsx-2783800609 LoginTopBarProfileButton"
        >
          <div
            className="jsx-2904874222 TopBarProfileMenu "
            onClick={[Function]}
          >
            <div
              className="jsx-2904874222 LoginTopBarProfileButton-name desktopOnly"
            />
            <div
              className="jsx-2904874222 LoginTopBarProfileButton-caret"
            />
          </div>
        </div>
      </div>
    </div>
  </header>
  <main
    className="jsx-4138488456"
  >
    <div
      className="c0"
    >
      <div
        className="c1"
      >
        <form
          method="GET"
          onSubmit={[Function]}
        >
          <div
            className="form-group form-group-lg"
          >
            <label
              className="h1 control-label"
              htmlFor="search"
            >
              Search Open Collective
            </label>
            <div
              className="c2"
            >
              <input
                className="c3 form-control"
                defaultValue=""
                id="search"
                name="q"
                placeholder="open source"
                type="search"
              />
              <button
                className="jsx-2820878800 jsx-219750138 Button blue c4"
                onClick={[Function]}
                type="submit"
              >
                <span
                  className="fa fa-search"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
      <div
        className="c5"
      >
        <div
          className="c6"
        >
          <svg
            fill="#46B0ED"
            height={100}
            viewBox="0 0 105 105"
            width={100}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12.5"
              cy="12.5"
              r="12.5"
            >
              <animate
                attributeName="fill-opacity"
                begin="0s"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="1;.2;1"
              />
            </circle>
            <circle
              cx="12.5"
              cy="52.5"
              fillOpacity=".5"
              r="12.5"
            >
              <animate
                attributeName="fill-opacity"
                begin="100ms"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="1;.2;1"
              />
            </circle>
            <circle
              cx="52.5"
              cy="12.5"
              r="12.5"
            >
              <animate
                attributeName="fill-opacity"
                begin="300ms"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="1;.2;1"
              />
            </circle>
            <circle
              cx="52.5"
              cy="52.5"
              r="12.5"
            >
              <animate
                attributeName="fill-opacity"
                begin="600ms"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="1;.2;1"
              />
            </circle>
            <circle
              cx="92.5"
              cy="12.5"
              r="12.5"
            >
              <animate
                attributeName="fill-opacity"
                begin="800ms"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="1;.2;1"
              />
            </circle>
            <circle
              cx="92.5"
              cy="52.5"
              r="12.5"
            >
              <animate
                attributeName="fill-opacity"
                begin="400ms"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="1;.2;1"
              />
            </circle>
            <circle
              cx="12.5"
              cy="92.5"
              r="12.5"
            >
              <animate
                attributeName="fill-opacity"
                begin="700ms"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="1;.2;1"
              />
            </circle>
            <circle
              cx="52.5"
              cy="92.5"
              r="12.5"
            >
              <animate
                attributeName="fill-opacity"
                begin="500ms"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="1;.2;1"
              />
            </circle>
            <circle
              cx="92.5"
              cy="92.5"
              r="12.5"
            >
              <animate
                attributeName="fill-opacity"
                begin="200ms"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="1;.2;1"
              />
            </circle>
          </svg>
        </div>
      </div>
    </div>
  </main>
  <div
    className="jsx-568600484 Footer"
  >
    <div
      className="jsx-568600484 content"
    >
      <div
        className="jsx-568600484"
      >
        <object
          className="jsx-568600484 logo"
          data="/static/images/opencollectivelogo-footer.svg"
          height="20"
          type="image/svg+xml"
        />
      </div>
      <div
        className="jsx-568600484"
      >
        <nav
          className="jsx-568600484"
        >
          <ul
            className="jsx-568600484"
          >
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="/learn-more"
              >
                How It Works
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://opencollective.com/#opensource"
              >
                Use Cases
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://opencollective.com/faq"
              >
                FAQ
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://forum.opencollective.com"
              >
                Forum
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://medium.com/open-collective"
              >
                Blog
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                href="/tos"
                onClick={[Function]}
              >
                Terms Of Service
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://opencollective.com/about"
              >
                About
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="mailto:info@opencollective.com"
              >
                Contact
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</div>
`;

exports[`Search Page renders search results with pagination 1`] = `
.c1 {
  box-sizing: border-box;
  width: 100%;
}

.c2 {
  box-sizing: border-box;
  margin-top: 16px;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c5 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  box-sizing: border-box;
  margin-left: 8px;
  margin-right: 8px;
  margin-top: 16px;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3.c3.c3 {
  border: none;
  border-bottom: 2px solid #46B0ED;
  border-radius: 0;
  box-shadow: none;
  display: block;
  font-family: lato;
  height: 3.4rem;
  padding: 0;
}

.c4.c4 {
  padding: 0 2rem;
}

.c0 {
  box-sizing: border-box;
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  max-width: 1200px;
}

@media screen and (min-width:40em) {
  .c5 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media screen and (min-width:52em) {
  .c5 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    width: 85%;
  }
}

<div>
  <header
    className="jsx-3598562734"
  >
    <div
      className="jsx-3598562734"
      id="top"
    />
    <div
      className="jsx-3593811691 loading TopBar"
    >
      <a
        className="jsx-3593811691"
        href="/"
        title="Go to Open Collective Homepage"
      >
        <img
          alt="Open Collective logo"
          className="jsx-3593811691 logo"
          height="40"
          src="/static/images/opencollective-icon.svg"
          width="40"
        />
      </a>
      <div
        className="jsx-3593811691 nav"
      >
        <ul
          className="jsx-3593811691 mediumScreenOnly"
        >
          <li
            className="jsx-3593811691"
          >
            <a
              className="jsx-3593811691 menuItem"
              href="/learn-more"
            >
              <span>
                How it works
              </span>
            </a>
          </li>
          <li
            className="jsx-3593811691"
          >
            <a
              className="jsx-3593811691 menuItem"
              href="/discover"
            >
              <span>
                discover
              </span>
            </a>
          </li>
          <li
            className="jsx-3593811691"
          >
            <a
              className="jsx-3593811691 menuItem"
              href="https://medium.com/open-collective"
            >
              <span>
                Blog
              </span>
            </a>
          </li>
        </ul>
        <div
          className="jsx-3593811691 separator"
        />
        <div
          className="jsx-2783800609 LoginTopBarProfileButton"
        >
          <div
            className="jsx-2904874222 TopBarProfileMenu "
            onClick={[Function]}
          >
            <div
              className="jsx-2904874222 LoginTopBarProfileButton-name desktopOnly"
            />
            <div
              className="jsx-2904874222 LoginTopBarProfileButton-caret"
            />
          </div>
        </div>
      </div>
    </div>
  </header>
  <main
    className="jsx-4138488456"
  >
    <div
      className="c0"
    >
      <div
        className="c1"
      >
        <form
          method="GET"
          onSubmit={[Function]}
        >
          <div
            className="form-group form-group-lg"
          >
            <label
              className="h1 control-label"
              htmlFor="search"
            >
              Search Open Collective
            </label>
            <div
              className="c2"
            >
              <input
                className="c3 form-control"
                defaultValue="Test"
                id="search"
                name="q"
                placeholder="open source"
                type="search"
              />
              <button
                className="jsx-2820878800 jsx-219750138 Button blue c4"
                onClick={[Function]}
                type="submit"
              >
                <span
                  className="fa fa-search"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
      <div
        className="c5"
      >
        <div
          className="c6"
        >
          <a
            href="/test-collective"
            onClick={[Function]}
          >
            <div
              className="jsx-3446301333 CollectiveCard undefined"
            >
              <div
                className="jsx-3446301333 head"
              >
                <div
                  className="jsx-3446301333 background"
                  style={Object {}}
                />
                <div
                  className="jsx-3446301333 logo"
                >
                  <div
                    className="jsx-1454345377 Logo"
                    style={
                      Object {
                        "backgroundImage": "url(/static/images/default-organization-logo.png)",
                        "height": 65,
                        "minWidth": 32.5,
                      }
                    }
                  >
                    <img
                      className="jsx-1454345377 logo"
                      src="/static/images/default-organization-logo.png"
                      style={
                        Object {
                          "maxHeight": 65,
                        }
                      }
                    />
                  </div>
                </div>
              </div>
              <div
                className="jsx-3446301333 body"
              >
                <div
                  className="jsx-3446301333 name"
                />
                <div
                  className="jsx-3446301333 description"
                />
              </div>
              <div
                className="jsx-3446301333 footer"
              />
            </div>
          </a>
        </div>
      </div>
      <div
        className="c7"
      >
        <ul
          className="pagination"
        >
          <li
            className="active"
          >
            <a
              href="?limit=20&offset=0&q=Test"
              onClick={[Function]}
            >
              1
            </a>
          </li>
          <li
            className=""
          >
            <a
              href="?limit=20&offset=20&q=Test"
              onClick={[Function]}
            >
              2
            </a>
          </li>
          <li
            className=""
          >
            <a
              href="?limit=20&offset=40&q=Test"
              onClick={[Function]}
            >
              3
            </a>
          </li>
          <li
            className=""
          >
            <a
              href="?limit=20&offset=60&q=Test"
              onClick={[Function]}
            >
              4
            </a>
          </li>
          <li
            className=""
          >
            <a
              href="?limit=20&offset=80&q=Test"
              onClick={[Function]}
            >
              5
            </a>
          </li>
        </ul>
      </div>
    </div>
  </main>
  <div
    className="jsx-568600484 Footer"
  >
    <div
      className="jsx-568600484 content"
    >
      <div
        className="jsx-568600484"
      >
        <object
          className="jsx-568600484 logo"
          data="/static/images/opencollectivelogo-footer.svg"
          height="20"
          type="image/svg+xml"
        />
      </div>
      <div
        className="jsx-568600484"
      >
        <nav
          className="jsx-568600484"
        >
          <ul
            className="jsx-568600484"
          >
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="/learn-more"
              >
                How It Works
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://opencollective.com/#opensource"
              >
                Use Cases
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://opencollective.com/faq"
              >
                FAQ
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://forum.opencollective.com"
              >
                Forum
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://medium.com/open-collective"
              >
                Blog
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                href="/tos"
                onClick={[Function]}
              >
                Terms Of Service
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="https://opencollective.com/about"
              >
                About
              </a>
            </li>
            <li
              className="jsx-568600484"
            >
              <a
                className="jsx-568600484"
                href="mailto:info@opencollective.com"
              >
                Contact
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
